test/res.location.js

```diff
@@ -58,7 +58,7 @@ describe('res', function(){
       });
 
       request(app)
-        .get('/?q=http://google.com\\@apple.com')
+        .get('/?q=http://google.com' + encodeURIComponent('\\@apple.com'))
```

Comment:
I know you tested locally, but I have no idea why this works, I guess we are converting the encoded @ (`%40`) back  to a @ before location setting? A percent encoded @ should not trigger this vuln Im pretty sure.

Reply:
No idea, do we need to know? It fixes it afaict in all supported versions.

Reply:
I shouldn't say no idea. I mean that I looked and it appears old supertest does not encode correctly (not at all IIRC). Encoding ourselves does it and new super test doesnt double escape or anything. This is not something I can spend more time so if you think we need to debug `supertest` to figure out why then go for it and approve when you are done.

Reply:
There is a bug report because we did not have test coverage of accepting `URL` instances. You requested changes on this. Can you approve it instead? I would like to merge this so we can also put out #5554

Reply:
I do want to understand why this still passes the test, but I wont block on it bc it is test only changes.